plasmapy/particles/exceptions.py

```diff
@@ -1,41 +1,61 @@
 """
-Collection of `Exceptions` and `Warnings` for PlasmaPy particles.
+Collection of exceptions and warnings for `plasmapy.particles`.
 """
 __all__ = [
-    "AtomicError",
-    "AtomicWarning",
+    "ParticleError",
+    "ParticleTypeError",
+    "ParticleValueError",
+    "ParticleWarning",
     "ChargeError",
     "InvalidElementError",
     "InvalidIonError",
     "InvalidIsotopeError",
     "InvalidParticleError",
-    "MissingAtomicDataError",
-    "MissingAtomicDataWarning",
+    "MissingParticleDataError",
+    "MissingParticleDataWarning",
     "UnexpectedParticleError",
 ]
 
 from plasmapy.utils import PlasmaPyError, PlasmaPyWarning
 
 
-class AtomicError(PlasmaPyError):
-    """An exception for errors in the `~plasmapy.particles` subpackage."""
+class ParticleError(PlasmaPyError):
+    """Base exception for errors in the `~plasmapy.particles` subpackage."""
 
     pass
 
 
-class MissingAtomicDataError(AtomicError):
+class ParticleTypeError(ParticleError, TypeError):
+    """
+    An exception for errors in the `~plasmapy.particles` subpackage
+    caused because an incorrect type was provided.
+    """
+
+    pass
+
+
+class ParticleValueError(ParticleError, ValueError):
+    """
+    An exception for errors in the `~plasmapy.particles` subpackage
+    caused because an incorrect value was provided.
+    """
+
+    pass
+
+
+class MissingParticleDataError(ParticleError):
     """An exception for missing atomic or particle data."""
 
     pass
```

Comment:
Could this also inherit from `AttributeError`?  I'm guessing this for the case when you're trying to access `particle.charge` and the attribute does not exist.  Or, is it for when the particle data is missing from the `json` files?

Reply:
It's mostly the latter case.  In some cases where this is used, the attribute in question still exists.

Reply:
Ok, I think the docstring should be updated to reflect this error is raised when the requested data as not found in the "databases."

Reply:
How about something like...

```suggestion
class MissingParticleDataError(ParticleError):
    """
    An exception for missing atomic or particle data in `plasmapy`'s 
    particle databases.
    """

    pass
```

Reply:
I made a change that was similar to that.